### PR TITLE
Remove last vestiges of nidump now that NGEN is dead

### PR DESCRIPTION
**Pull request by jkoritzinsky**
After the some of the mass cleanup PRs, some dead code from nidump was left around. This PR removes what is left.

### CODE REVIEW

File: src/coreclr/debug/daccess/gcdump_dac.cpp [hunk 1]
```diff
@@ -0,0 +1,48 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include <stdafx.h>
+
+/* There is no DAC build of gcdump, so instead
```
Comment (elinor-fung) [inline src/coreclr/debug/daccess/gcdump_dac.cpp:6]: You mean you didn't think `REVISIT_TODO Mon 10/10/2005` from nidump.cpp was useful to keep?

**Review by elinor-fung — APPROVED** [review]
cc @dotnet/dotnet-diag

File: src/coreclr/debug/daccess/gcdump_dac.cpp (hunk 1 above)
Comment (jkoritzinsky) [reply src/coreclr/debug/daccess/gcdump_dac.cpp:6]: I figured if it wasn't fixed by now, it's not going to be fixed haha

File: src/coreclr/vm/method.hpp [hunk 2]
```diff
@@ -2189,10 +2186,7 @@ class MulticoreJitPrepareCodeConfig : public PrepareCodeConfig
 class MethodDescChunk
 {
     friend class MethodDesc;
-    friend class CheckAsmOffsets;
-#ifdef DACCESS_COMPILE
-    friend class NativeImageDumper;
-#endif // DACCESS_COMPILE
+    friend class CheckAsmOffsets; // DACCESS_COMPILE
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/vm/method.hpp:2189]: Trailing `DACCESS_COMPILE`?

**Review by AaronRobinsonMSFT — APPROVED** [review]

File: src/coreclr/vm/method.hpp (hunk 2 above)
Comment (jkoritzinsky) [reply src/coreclr/vm/method.hpp:2189]:
```suggestion
    friend class CheckAsmOffsets;
```

**Pull request merged**